cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/CompositeDatasetDefinition.java

```diff
@@ -116,10 +117,9 @@ protected CompositeDatasetDefinition(String name,
 
   @Override
   public DatasetSpecification configure(String instanceName, DatasetProperties properties) {
-    List<DatasetSpecification> specs = new ArrayList<>();
-    for (Map.Entry<String, ? extends DatasetDefinition> impl : this.delegates.entrySet()) {
-      specs.add(impl.getValue().configure(impl.getKey(), properties));
-    }
+    List<DatasetSpecification> specs = this.delegates.entrySet().stream()
```

Comment:
was this code transformation done by the IDE?

Reply:
This one yes. IDE would suggest a change once you set it to use Java 8.